### ISSUE TITLE
Does WScaleLayer in layer.py change learnable weights or just output tensor?

### ISSUE THREAD

**Issue by gaoming96** [Latope2-150/Progressive_Growing_of_GANs-PyTorch#3]
Hi Jeromerony, thanks so much for this great project, it works really well. 

I have a question about your `WScaleLayer` class in `layer.py`. Is this class used to normalize the input tensor? However, both the original paper and the name seem to change the learnable weights directly (if I understand it correctly)?

Thanks in advance.

**Comment by jeromerony** [comment]
Hi, thanks!
So it is named WScale because of the original name in the tensorflow repo.
This class normalizes the input tensor instead of the weights since it is more convenient in PyTorch. Mathematically, it is exactly the same.

**Comment by gaoming96** [comment]
Oh, I get it. Thank you!

**Comment by iperov** [comment]
looks like it's not trainable, so it has no effect
